src/Lean/PrettyPrinter/Delaborator/Builtins.lean

```diff
@@ -187,16 +187,28 @@ def shouldShowMotive (motive : Expr) (opts : Options) : MetaM Bool := do
   <||> (← getPPMotivesPi opts <&&> returnsPi motive)
   <||> (← getPPMotivesNonConst opts <&&> isNonConstFun motive)
 
+partial def withMDatas [Inhabited α] (x : DelabM α) : DelabM α := do
```

Comment:
Should we extract a function `withMData` without recursion and use it in `delabMData`?

Reply:
Good idea. Will fix.